reserved_ipv6.go

```diff
@@ -0,0 +1,139 @@
+package godo
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+const resourceV6Type = "ReservedIPv6"
+const reservedIPV6sBasePath = "v2/reserved_ipv6"
+
+// ReservedIPV6sService is an interface for interfacing with the reserved IPV6s
+// endpoints of the Digital Ocean API.
+type ReservedIPV6sService interface {
+	List(context.Context, *ListOptions) ([]ReservedIPV6, *Response, error)
+	Get(context.Context, string) (*ReservedIPV6, *Response, error)
+	Reserve(context.Context, *ReservedIPV6ReserveRequest) (*ReservedIPV6, *Response, error)
+	Release(context.Context, string) (*Response, error)
+}
+
+// ReservedIPV6sServiceOp handles communication with the reserved IPs related methods of the
+// DigitalOcean API.
+type ReservedIPV6sServiceOp struct {
+	client *Client
+}
+
+var _ ReservedIPV6sService = &ReservedIPV6sServiceOp{}
+
+// ReservedIPV6 represents a Digital Ocean reserved IP.
+type ReservedIPV6 struct {
+	RegionSlug string    `json:"region_slug"`
+	IP         string    `json:"ip"`
+	ReservedAt time.Time `json:"reserved_at"`
+	Droplet    *Droplet  `json:"droplet,omitempty"`
+}
+
+func (f ReservedIPV6) String() string {
+	return Stringify(f)
+}
+
+// URN returns the reserved IP in a valid DO API URN form.
+func (f ReservedIPV6) URN() string {
+	return ToURN(resourceV6Type, f.IP)
+}
+
+type reservedIPV6sRoot struct {
+	ReservedIPs []ReservedIPV6 `json:"reserved_ips"`
+	Links       *Links         `json:"links"`
+	Meta        *Meta          `json:"meta"`
+}
+
+type reservedIPV6Root struct {
+	ReservedIP *ReservedIPV6 `json:"reserved_ip"`
+	Links      *Links        `json:"links,omitempty"`
+}
+
+// ReservedIPV6ReserveRequest represents a request to reserve a reserved IP.
+type ReservedIPV6ReserveRequest struct {
+	Region string `json:"region_slug,omitempty"`
+}
+
+// List all reserved IPV6s.
+func (r *ReservedIPV6sServiceOp) List(ctx context.Context, opt *ListOptions) ([]ReservedIPV6, *Response, error) {
+	path := reservedIPV6sBasePath
+	path, err := addOptions(path, opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := r.client.NewRequest(ctx, http.MethodGet, path, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	root := new(reservedIPV6sRoot)
+	resp, err := r.client.Do(ctx, req, root)
+	if err != nil {
+		return nil, resp, err
```

Comment:
Curious if `err == nil` what is to return in `resp` 🤔 ?

Reply:
it will be nil only